### DISCUSSION TITLE
Specify dependency type for depcheck profile capabilities?

### DISCUSSION THREAD

**Discussion by simarmol** [microsoft/rnx-kit#1590] [Q&A] [answered]
Hi, I was looking for a way to declare something as a devdependency in the capabilities but didn't see it mentioned in the [docs](https://github.com/microsoft/rnx-kit/tree/main/packages/dep-check#custom-profiles). However, looking around in the profiles directory I eventually found a devOnly field?

https://github.com/microsoft/rnx-kit/blob/2e044504eccd83a0b90349bba6bcf6d3c80179d4/packages/dep-check/src/profiles/profile-0.61.ts#L25

I think it should be mentioned in the main depcheck docs, but I was also wondering if there are comparable properties for peer dependencies, or there's no need because that's determined by the kit type of app vs library? The default is prod dependencies for an 'app', with devDependencies indicated by the devOnly property. 

**Questions**
1. Is anything that doesn't have 'devOnly' declared written as a peer and devDependency for the 'library' type kit?
2. How would you declare a direct dependency for a 'library' kit rather, if non-dev dependencies in profiles are written as peer/dev to the library-kit's package.json?
3. I noticed that devOnly is a property for a meta capability as well for capabilities. Does it overwrite its children's declarations of devOnly if the value is present in the meta capability?
4. I read the depcheck doc and the design doc, which confused me more because it implied you wouldn't want to use peer dependencies at all, since they aren't enforced and instead use depcheck to keep direct dependency versions synced instead. But that doesn't seem to be the behavior of depcheck?

Let me know if this should be in discussions? I did have one actionable item (adding to the depcheck readme) so I started writing it in issues.

**Comment by tido64** [comment, marked as answer]
1. Yes, dependencies are either declared under `dependencies` for apps, or `peerDependencies` + `devDependencies` for libraries.
2. You typically don't. You want dep-check to manage that for you so that you don't accidentally end up with duplicates. If you need to do this for some reason, the solution would be to not have dep-check manage it.
3. Not quite. The `devOnly` flag controls whether a capability is considered depending on whether you are the author or the consumer of the package. If you're the author, dep-check resolves it to real packages and makes sure the resolved packages are correctly declared. If you're a consumer, the capability is ignored.
4. If you have any suggestions on how to improve the docs, please feel free to submit PRs. The intention of the point regarding `peerDependencies` is that we find it lacking in terms of enforcing versions on the consumer side.
